src/writer.rs

```diff
@@ -74,7 +95,16 @@ impl<W: Write> Writer<W> {
 
     /// Writes the given event to the underlying writer.
     pub fn write_event<'a, E: AsRef<Event<'a>>>(&mut self, event: E) -> Result<usize> {
-        match *event.as_ref() {
+        if self.indent_size > 0 {
+            match *event.as_ref() {
+                Event::End(_) => {
+                    let new_len = self.indents.len() - self.indent_size;
+                    self.indents.truncate(new_len);
```

Comment:
Not sure it is a good idea to truncate it because it will very likely grow for the next node.
I guess it could be better to use write `&indent[..indent_len]` directly and reduce the amount of `extend`/`truncate`.